Source/WebCore/platform/mediastream/mac/CoreAudioCaptureDeviceManager.cpp

```diff
@@ -102,7 +102,12 @@ static bool deviceHasOutputStreams(AudioObjectID deviceID)
     return deviceHasStreams(deviceID, address);
 }
 
-static bool isValidCaptureDevice(const CoreAudioCaptureDevice& device, bool filterTapEnabledDevices)
+static bool isVirtualDeviceFromLabel(const String& label)
+{
+    return label.contains("WebexMediaAudioDevice"_s) || label.contains("Microsoft Teams Audio"_s);
```

Comment:
I have a few thoughts:
1. I wonder if this should go in BrowserEngineKit instead of WebKit since other browsers likely want to apply the same filtering.
2. Is there a way for AVAudioSession to do this filtering for us so that we don't have to maintain a hardcoded deny list? Since other parts of the system filter out these devices (e.g., Control Center), it seems like this logic already exists elsewhere.

Reply:
This is macOS only so does not apply to AVAudioSession/BEK.
macOS system apps like sound settings app are doing filtering.

This only seems to be feasible in GPU process where we are doing capture/audio rendering and not UIProcess where we are doing enumeration.

We could try to move enumeration to UIProcess but it is also unclear whether this filtering would be too harsh, Alexei mentioned OBS for instance.